### PR TITLE
[3.x] Bind NOTIFICATION_LOCAL_TRANFORM_CHANGED

### PR DESCRIPTION
**Pull request by jasonwinterpixel**
The NOTIFICATION_LOCAL_TRANFORM_CHANGED constant is not exposed to GDScript, which means that using `canvas_item.set_notify_local_transform` requires hard coding '35' into your GDScript.

This PR binds this constant to GDScript.

Also improved the documentation for the associated methods.

master branch version: https://github.com/godotengine/godot/pull/59352

### CODE REVIEW

File: doc/classes/CanvasItem.xml [hunk 1]
```diff
@@ -384,14 +384,14 @@
 			<return type="void" />
 			<argument index="0" name="enable" type="bool" />
 			<description>
-				If [code]enable[/code] is [code]true[/code], children will be updated with local transform data.
+				If [code]enable[/code] is [code]true[/code], this node will receive [constant NOTIFICATION_LOCAL_TRANSFORM_CHANGED] when it's local transform changes.
```
Comment (Calinou) [inline doc/classes/CanvasItem.xml:387]:
```suggestion
				If [code]enable[/code] is [code]true[/code], this node will receive [constant NOTIFICATION_LOCAL_TRANSFORM_CHANGED] when its local transform changes.
```

Comment (Calinou) [inline doc/classes/CanvasItem.xml:387]: Possessive "its" doesn't have an apostrophe:

```suggestion
				If [code]enable[/code] is [code]true[/code], this node will receive [constant NOTIFICATION_LOCAL_TRANSFORM_CHANGED] when its local transform changes.
```

File: doc/classes/CanvasItem.xml [hunk 2]
```diff
@@ -384,14 +384,14 @@
...
 			</description>
 		</method>
 		<method name="set_notify_transform">
 			<return type="void" />
 			<argument index="0" name="enable" type="bool" />
 			<description>
-				If [code]enable[/code] is [code]true[/code], children will be updated with global transform data.
+				If [code]enable[/code] is [code]true[/code], this node will receive [constant NOTIFICATION_TRANSFORM_CHANGED] when it's global transform changes.
```
Comment (Calinou) [inline doc/classes/CanvasItem.xml:394]:
```suggestion
				If [code]enable[/code] is [code]true[/code], this node will receive [constant NOTIFICATION_TRANSFORM_CHANGED] when its global transform changes.
```

**Review by Calinou — DISMISSED** [review]

**Comment by jasonwinterpixel** [comment]
Good catch. Fixed.

**Review by akien-mga — APPROVED** [review]

**Pull request merged**